internal/testutils/daemon.go

```diff
@@ -133,6 +141,12 @@ paths:
 	stopped = make(chan struct{})
 	processPid := make(chan int)
 	go func() {
+		// When using a shared daemon we should not log using the test function.
```

Comment:
> If we're using a single shared daemon for all the tests the t* instance
may refer to another test that has been over

Oh, I think we should indeed not use the `*testing.T` at all when the daemon is shared, including in `require.NoError` and similar, because those call `t.FailNow()` which causes the `*testing.T` instance to fail, so it can lead to the wrong test failing (I'm not sure what happens if that test has already finished, maybe then it doesn't cause any test to fail).

Reply:
Yeah, I was thinking the same... I delayed since it wasn't lead to races per se, but indeed potential false positives... I'd translate those to panics, although not sure if it's more helpful on failures

Reply:
Ok this is now handled